### PR TITLE
#1864 - PT Integration Files Review - e-Cert and Feedback Files

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
### Part-time e-Cert file
- Adjusted new header/record/footer record types from 3 to 2 characters.
- Changed the "sent title" using a new const `ECERT_PT_SENT_TITLE`.
- Added a new field for PD/PDD. Currently reading from the student profile. Later it should be determined from where it will be actually read.
- Reading country, province, and postal code in the same way as it is for full-time. This is still pending business answers about how it will properly work.
- Header record now includes the file sequence.

### Part-time feedback file
- Fixes for all the positions that were shifted by 1.
- Fix for the total "number of records" calculation.
- Replace the obsolete method `onConflict`.
- Removed no longer needed different codes for the feedback file.
- Fix an error preventing the e-Cert from being generated when some awards were not present.

### CODE REVIEW

File: sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/models/e-cert-integration-model.ts [hunk 1]
```diff
@@ -63,9 +69,6 @@ export enum RecordTypeCodes {
   ECertPartTimeHeader = "01",
   ECertPartTimeRecord = "02",
   ECertPartTimeFooter = "99",
-  ECertPartTimeFeedbackHeader = "001",
```
Comment (ann-aot) [inline sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/models/e-cert-integration-model.ts:66]: I guess `RecordTypeCodes` is used while creation and response reading right? If so, i think we need to adjust the comment line 57

File: sources/packages/backend/libs/integrations/src/services/disbursement-schedule/disbursement-schedule.service.ts [hunk 2]
```diff
@@ -18,14 +18,17 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
...
+   * Get the disbursement schedule by its document number.
+   * @param documentNumber document number.
+   * @returns disbursement schedule respective to the provided document number.
    */
   async getDisbursementScheduleByDocumentNumber(
     documentNumber: number,
   ): Promise<DisbursementSchedule> {
-    return this.repo.findOne({ where: { documentNumber } });
+    return this.repo.findOne({
+      select: { id: true },
```
Comment (ann-aot) [inline sources/packages/backend/libs/integrations/src/services/disbursement-schedule/disbursement-schedule.service.ts:29]: 👍

File: sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/models/e-cert-integration-model.ts (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/models/e-cert-integration-model.ts:66]: Comment adjusted.

**Review by ann-aot — APPROVED** [review]
LGTM 👍 Good work @andrewsignori-aot

File: sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts [hunk 3]
```diff
@@ -268,17 +269,20 @@ export class ECertFileHandler extends ESDCFileHandler {
...
       addressLine2: addressInfo.addressLine2,
       city: addressInfo.city,
       country: addressInfo.country,
       provinceState: addressInfo.provinceState,
       postalCode: addressInfo.postalCode,
-      email: application.student.user.email,
-      gender: application.student.gender,
+      email: student.user.email,
+      gender: student.gender,
+      // TODO: getting the information directly from the student profile since there is no
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts:283]: 👍

File: sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-footer.ts [hunk 4]
```diff
@@ -25,15 +25,14 @@ export class ECertPartTimeFileFooter extends ECertFileFooter {
...
   createFromLine(line: string): ECertPartTimeFileFooter {
     const footer = new ECertPartTimeFileFooter();
-    footer.recordTypeCode = line.substring(0, 3) as RecordTypeCodes;
-    // Here total record count is the total records.
-    footer.recordCount =
-      parseInt(line.substring(43, 52)) + parseInt(line.substring(52, 61));
-    footer.totalSINHash = parseInt(line.substring(61, 76));
+    footer.recordTypeCode = line.substring(0, 2) as RecordTypeCodes;
+    // Total number of records not including header and footer.
+    footer.recordCount = +line.substring(51, 60);
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-footer.ts:30]: Is line.substring(43, 52) not included here is about not including header and footer.

File: sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-header.ts [hunk 5]
```diff
@@ -20,16 +21,17 @@ export class ECertPartTimeFileHeader extends ECertFileHeader {
     const header = new StringBuilder();
     header.append(this.recordTypeCode);
     header.appendWithEndFiller(ORIGINATOR_CODE, 4, SPACE_FILLER);
-    header.appendWithEndFiller(ECERT_SENT_TITLE, 40, SPACE_FILLER);
+    header.appendWithEndFiller(ECERT_PT_SENT_TITLE, 40, SPACE_FILLER);
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-header.ts:24]: 👍

File: sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-footer.ts (hunk 4 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-footer.ts:30]: Not really, I was not able to understand what was the idea from the previous implementation.
`line.substring(43, 52)` represents the total of records that were successfully processed. The successfully processed records are not part of the file so they should not be part of the total records validation.
image

File: sources/packages/backend/libs/utilities/src/esdc-utils.ts [hunk 6]
```diff
@@ -65,3 +65,12 @@ export function getPartTimeMaritalStatusCode(
   }
   return maritalStatus === RelationshipStatus.Single ? "SI" : "SP";
 }
+
+/**
+ * Get the e-Cert flag for the borrower's persistent or prolonged disability status.
+ * @param hasPDorPPD indicates if a borrower has a persistent or prolonged disability.
+ * @returns "Y" or "N" flag.
+ */
+export function getPPDFlag(hasPPD?: boolean): string {
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/utilities/src/esdc-utils.ts:74]: If hasPPD is not intended to be an optional field, we can ignore `?`. Just my opinion, feel free to ignore.

Comment (andrewsignori-aot) [reply sources/packages/backend/libs/utilities/src/esdc-utils.ts:74]: It is mandatory for the file but optional on SIMS, that is why it receives the optional parameter and converts it to a value.

File: sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-footer.ts (hunk 4 above)
Comment (dheepak-aot) [reply sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-footer.ts:30]: I see that is weird. We can ask the reason to the person who is dealing with the file integration if not already asked.

**Review by dheepak-aot — APPROVED** [review]
Nice work @andrewsignori-aot . LGTM.

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrewsignori-aot

**Review by andrepestana-aot — APPROVED** [review]
LGTM

File: sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-footer.ts (hunk 4 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-footer.ts:30]: I asked them for a sample file and I did the changes based on the file provided (that is aligned with the documentation), so it should be ok.
What is weird to me is how it worked or was even tested before, maybe some misdirection from previous documentation.
Either way, the upcoming tests will have the files validated.

**Pull request merged**